### DISCUSSION TITLE
Simple Program not executing

### DISCUSSION THREAD

**Discussion by timcad12** [PHPMailer/PHPMailer#3075] [General]
I have a very simple code that looks like this:
<?php

echo "1";
use PHPMailer\PHPMailer\PHPMailer;
echo "2";
use PHPMailer\PHPMailer\SMTP; 
echo "3";
use PHPMailer\PHPMailer\Exception; 
echo "4";
require 'PHPMailer/Exception.php'; 
echo "5";
require 'PHPMailer/PHPMailer.php'; 
echo "6";
require 'PHPMailer/SMTP.php'; 
echo "7";
?>

I use two different hosting companies - call them X and Y. When I upload and execute using X it returns 1234567 as expected. When I upload to to company Y and execute I get 12345 so the code never finished. The file structure for the PHPMailer folders are the same in both cases.

FYI company X is Inmotion Hosting and Y is Network Solutions.

**Comment by timcad12** [comment]
Sorry - here is the code:

<?php

echo "1";
use PHPMailer\PHPMailer\PHPMailer;
echo "2";
use PHPMailer\PHPMailer\SMTP; 
echo "3";
use PHPMailer\PHPMailer\Exception; 
echo "4";
require 'PHPMailer/Exception.php'; 
echo "5";
require 'PHPMailer/PHPMailer.php'; 
echo "6";
require 'PHPMailer/SMTP.php'; 
echo "7";
?>

**Comment by timcad12** [comment]
-<?php

echo "1";
use PHPMailer\PHPMailer\PHPMailer;
echo "2";
use PHPMailer\PHPMailer\SMTP; 
echo "3";
use PHPMailer\PHPMailer\Exception; 
echo "4";
require 'PHPMailer/Exception.php'; 
echo "5";
require 'PHPMailer/PHPMailer.php'; 
echo "6";
require 'PHPMailer/SMTP.php'; 
echo "7";
?>

**Reply by Synchro** [reply]
Likely your PHP include paths are configured differently, or you have incorrect ownership or permissions on the files. The bigger question though is why are you not using composer?

Incidentally I use Inmotion for a client, and I have to say that their hosting configuration is an absolute nightmare to work with, utterly bizarre.

**Comment by timcad12** [comment]
Thanks for the quick reply Synchro. Are you talking about the paths defined in this file. They are the same program run on the two different providers and the PHP folder structures are identical.

I don't use Composer because I use Dreamweaver to develop my websites and not Wordpress.

Couldn't agree more about NS!

**Reply by Synchro** [reply]
No, it's that PHP has a config setting called `include_path` that determines where PHP looks when you ask it to `include`/`require` a file, and it might be set differently in your two installations. Instead of just echoing those numbers, it would help to see the actual errors that are being generated because I'd guess they are saying either "file not found" , or "permission denied". You should be able to see the errors if you run this script on a command line, enable the display of error messages in your PHP config, or look in your web server's error log.

Composer doesn't have anything to do with DreamWeaver ("Now that's a name I've not heard in a long time...") or WordPress; you can use composer independently of whatever tools you're using, and it will stop you having to worry about loading PHP files.

**Comment by timcad12** [comment]
Yea that is one of the frustrating things about NS. No file manager, not error reports generated by this so there is just no feedback at all. It just stops executing. 

Will look into Composer. 

I still like Dreamweaver...have worked with it for 14 years.

**Reply by Synchro** [reply]
You can enable error reporting in your script. Add this line at the top:

    ini_set('display_errors', true);

**Comment by timcad12** [comment]
Here is what I get: Parse error: syntax error, unexpected '[' in /data/25/2/152/8/2478008/user/2713785/htdocs/PHPMailer/PHPMailer.php on line 167

**Reply by Synchro** [reply]
I would guess that means you're running a *very* old version of PHP that doesn't support short array syntax, which was introduced in PHP 5.4.

**Comment by timcad12** [comment]
Well isn't that a function of the Network Solutions web hosting package? Dreamweaver does not throw that error in "Live" mode.

**Comment by timcad12** [comment]
Aha!!!!!

PHP version: 5.3.29-pl0-gentoo12345
Parse error: syntax error, unexpected '[' in /data/25/2/152/8/2478008/user/2713785/htdocs/PHPMailer/PHPMailer.php on line 167

**Reply by Synchro** [reply]
Ouch. That version of PHP is _10 years_ out of date. Your web hosting service is terrible!

**Comment by timcad12** [comment]
Yep. But I found where to select the newer version so I selected 8.2. And that fixed the problem! Thank you so much for the help. I never would have suspected that! You're a genius!!!

**Comment by timcad12** [comment]
So now I have made my way through the whole form (not the short version I was testing with)  and it balks on the final step of actually sending the email.  After the if(!$mail->send()) {...  step there is a pause for about 45 seconds and then I get a blank screen with "504 Gateway Time-out openresty/1.25.3.1.  You can try it at http://www.bscontractorsllc.com/Test1.php. Just put in a made up name, email address and phone number and hit submit

**Reply by Synchro** [reply]
Sounds like your hosting provider is blocking outbound SMTP, which is so common that it's covered extensively [in the docs](https://github.com/PHPMailer/PHPMailer/wiki/Troubleshooting#smtp-error-could-not-connect-to-smtp-host).
